site/src/xServices/workspaceSchedule/workspaceScheduleXService.ts

```diff
@@ -50,160 +53,227 @@ export type WorkspaceScheduleEvent =
+            onDone: [
+              {
+                actions: ["assignPermissions"],
+                target: "gettingTemplate",
+              },
+            ],
+            onError: [
+              {
+                actions: "assignGetPermissionsError",
+                target: "error",
+              },
+            ],
           },
+          tags: "loading",
         },
-        tags: "loading",
-      },
-      gettingPermissions: {
-        entry: "clearGetPermissionsError",
-        invoke: {
-          src: "checkPermissions",
-          id: "checkPermissions",
-          onDone: [
-            {
-              actions: ["assignPermissions"],
+        gettingTemplate: {
+          entry: "clearGetTemplateError",
+          invoke: {
+            src: "getTemplate",
+            id: "getTemplate",
+            onDone: {
               target: "presentForm",
+              actions: ["assignTemplate"],
             },
-          ],
-          onError: [
-            {
-              actions: "assignGetPermissionsError",
+            onError: {
               target: "error",
+              actions: ["assignGetTemplateError"],
             },
-          ],
-        },
-      },
-      presentForm: {
-        on: {
-          SUBMIT_SCHEDULE: "submittingSchedule",
+          },
+          tags: "loading",
         },
-      },
-      submittingSchedule: {
-        invoke: {
-          src: "submitSchedule",
-          id: "submitSchedule",
-          onDone: {
-            target: "submitSuccess",
-            actions: "displaySuccess",
+        presentForm: {
+          on: {
+            SUBMIT_SCHEDULE: {
+              target: "submittingSchedule",
+              actions: "assignAutoStopChanged",
+            },
           },
-          onError: {
-            target: "presentForm",
-            actions: ["assignSubmissionError"],
+        },
+        submittingSchedule: {
+          invoke: {
+            src: "submitSchedule",
+            id: "submitSchedule",
+            onDone: [
+              {
+                cond: "autoStartChanged",
+                target: "showingRestartDialog",
+              },
+              { target: "done" },
+            ],
+            onError: {
+              target: "presentForm",
+              actions: ["assignSubmissionError"],
+            },
           },
+          tags: "loading",
         },
-        tags: "loading",
-      },
-      submitSuccess: {
-        on: {
-          SUBMIT_SCHEDULE: "submittingSchedule",
+        showingRestartDialog: {
+          on: {
+            RESTART_WORKSPACE: {
+              target: "done",
+              actions: "restartWorkspace",
+            },
+            APPLY_LATER: "done",
+          },
         },
-      },
-      error: {
-        on: {
-          GET_WORKSPACE: "gettingWorkspace",
+        error: {
+          on: {
+            GET_WORKSPACE: "gettingWorkspace",
+          },
         },
+        done: {},
```

Comment:
Is this a 'finalState'? Should we label it as such? https://xstate.js.org/docs/guides/final.html#api